metricflow/test/integration/test_configured_cases.py

```diff
@@ -66,8 +66,20 @@ def render_time_constraint(
         stop_time_plus_one_day = (
             datetime.datetime.strptime(stop_time, time_format) + datetime.timedelta(days=1)
         ).strftime(time_format)
+
         stop_expr = self.cast_to_ts(f"{stop_time_plus_one_day}")
-        return f"{expr} >= {start_expr} AND {expr} < {stop_expr}"
+        return f"{self.cast_expr_to_ts(expr)} >= {start_expr} AND {self.cast_expr_to_ts(expr)} < {stop_expr}"
+
+    def render_between_time_constraint(
+        self,
+        expr: str,
+        start_time: str,
+        stop_time: str,
+    ) -> str:
+        """Render an expression like "ds between timestamp '2020-01-01' AND timestamp '2020-01-02'" since Trino require timestamp literals to be wrapped in a timestamp() function."""
```

Comment:
I think this is an old comment. Let's update to something more like this:
```suggestion
        """Render an expression like "ds between timestamp '2020-01-01' AND timestamp '2020-01-02'".
        
         This will cast the literals as needed for each engine Trino, and provide an alternative to incrementing
         the date as we do in render_time_constraint. Using BETWEEN is more robust for cases involving potentially
         mixed granularities.
         """
```

Reply:
Thanks for taking another look. I might get sometime tomorrow to address inline comments and try to handle rebase to pick relevant commits.

Reply:
@sarbmeetka if you do that'd be great, and please feel free to split the github action changes out to a separate PR (or just remove them altogether) - that way this one would be cleaned up for merge and I can mess around with the CI configs on top of it.

Reply:
Commit history should look cleaner now.

Reply:
I havent changed anything on github actions yet.